freshmaker/handlers/koji/rebuild_eus_base_images.py

```diff
@@ -0,0 +1,107 @@
+from freshmaker.lightblue import LightBlue
+from freshmaker.errata import Errata
+from .rebuild_images_on_rpm_advisory_change import RebuildImagesOnRPMAdvisoryChange
+
+
+class RebuildEUSBaseImages(RebuildImagesOnRPMAdvisoryChange):
+    """
+    Rebuilds Extended Life Support base images
+    """
+
+    name = 'RebuildEUSBaseImages'
+
+    def can_handle(self, event):
+        if not super().can_handle(event):
+            return False
+
+        # If 'content_sets' of the event doesn't exist yet, create and fill it
+        if self.event.content_sets is None:
+            errata = Errata()
+            self._set_event_content_sets(errata, int(event.search_key))
+
+        # Check if any of the content sets is EUS content set
+        for content_set in self.event.content_sets:
+            if re.match(r'.*-eus($|\W)', content_set):
+                return True
+
+        return False
+
+    def _find_images_to_rebuild(self, errata_id):
+        """
+        Finds docker rebuild images from each build added to specific Errata
+        advisory.
+
+        But only EUS base images that DON'T have children should be rebuilt.
+        So other bathes including EUS images with children will be handled by
+        other handler and hence should be filtered out here.
+
+        :param int errata_id: Errata ID.
+        """
+        errata = Errata()
+        errata_id = int(errata_id)
+
+        self._set_event_content_sets(errata)
+
+        self.log_info('RPMs from advisory ends up in following content sets: '
+                      '%s', self.event.content_sets)
+
+        # Query images from LightBlue by signed RPM's srpm name and found
+        # content sets
+        lb = LightBlue(server_url=conf.lightblue_server_url,
+                       cert=conf.lightblue_certificate,
+                       private_key=conf.lightblue_private_key,
+                       event_id=self.current_db_event_id)
+
+        # Limit the Lightblue query to particular leaf images if set in Event.
+        leaf_container_images = None
+        if isinstance(self.event, ManualRebuildWithAdvisoryEvent):
+            leaf_container_images = self.event.container_images
+
+        # Get binary rpm nvrs which are affected by the CVEs in this advisory
+        affected_nvrs = self.event.advisory.affected_rpm_nvrs
+
+        # If there is no CVE affected binary rpms, this can be non-RHSA advisory,
+        # just rebuild images that have the builds in this advisory installed
+        if not affected_nvrs:
+            affected_nvrs = errata.get_binary_rpm_nvrs(errata_id)
+
+        self.log_info(
+            "Going to find all the container images to rebuild as "
+            "result of %r update.", affected_nvrs)
+        batches = lb.find_images_to_rebuild(
+            affected_nvrs, self.event.content_sets,
+            filter_fnc=self._filter_out_not_allowed_builds,
+            published=None, release_categories=None,
```

Comment:
`published=False` should be set explicitly for this filter:
https://github.com/redhat-exd-rebuilds/freshmaker/blob/2f3bf7d8cf497ea9a216b391067d0ba4672f82f0/freshmaker/lightblue.py#L949-L956

I don't think it really matters, but it's probably best to be consistent here.